lib/tenstorrent/bh_arc/init.c

```diff
@@ -264,6 +265,30 @@ static void EthInit(void)
 	}
 }
 
+/**
+ * @brief Redo Tensix init that gets cleared on Tensix reset
+ *
+ * This includes all NOC programming and any programming within the tile.
+ */
+static uint8_t ReinitTensix(uint32_t msg_code, const struct request *req, struct response *rsp)
+{
+	ClearNocTranslation();
+	/* We technically don't have to re-program the entire NOC (only the Tensix NOC portions),
+	 * but it's simpler to reuse the same functions to re-program all of it.
+	 */
+	NocInit();
+	if (get_fw_table()->feature_enable.cg_en) {
+		EnableTensixCG();
+	}
+	if (get_fw_table()->feature_enable.noc_translation_en) {
+		InitNocTranslationFromHarvesting();
+	}
+
+	return 0;
+}
+
+REGISTER_MESSAGE(MSG_TYPE_REINIT_TENSIX, ReinitTensix);
```

Comment:
The build fails when building recovery firmware due to missing symbols. Repro with
```shell
west build -p auto -b tt_blackhole/tt_blackhole/smc app/smc -- -DCONFIG_TT_SMC_RECOVERY=y
```

This should fix it.
```suggestion
#ifndef CONFIG_TT_SMC_RECOVERY
/**
 * @brief Redo Tensix init that gets cleared on Tensix reset
 *
 * This includes all NOC programming and any programming within the tile.
 */
static uint8_t ReinitTensix(uint32_t msg_code, const struct request *req, struct response *rsp)
{
	ClearNocTranslation();
	/* We technically don't have to re-program the entire NOC (only the Tensix NOC portions),
	 * but it's simpler to reuse the same functions to re-program all of it.
	 */
	NocInit();
	if (get_fw_table()->feature_enable.cg_en) {
		EnableTensixCG();
	}
	if (get_fw_table()->feature_enable.noc_translation_en) {
		InitNocTranslationFromHarvesting();
	}

	return 0;
}

REGISTER_MESSAGE(MSG_TYPE_REINIT_TENSIX, ReinitTensix);
#endif
```

Reply:
Thanks, trying again.